libs/newsletter/src/driver/hubspot/newsletter.service.spec.ts

```diff
@@ -0,0 +1,139 @@
+
+import { Title } from '@angular/platform-browser';
+import { HttpClient, HttpRequest } from '@angular/common/http';
+import { DaffNewsletterHubspotService } from './newsletter.service';
+import { DOCUMENT } from '@angular/common';
+import { Router } from '@angular/router';
+import { RouterTestingModule } from '@angular/router/testing';
+
+describe('DaffNewsletterHubspotService', () => {
+  let newsletterService: DaffNewsletterHubspotService;
+  let httpMock: HttpTestingController;
+  let doc: Document;
+  let httpClient: HttpClient;
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      imports: [
+        HttpClientTestingModule,
+        RouterTestingModule,
+      ],
+      providers: [
+        DaffNewsletterHubspotService,
+        { provide: DaffNewsletterConfig, useValue: {} },
+        { provide: DaffNewsletterTransformer, useExisting: DaffNewsletterHubspotTransformer }
+      ]
+    });
+
+    httpMock = TestBed.get(HttpTestingController);
+    newsletterService = TestBed.get(DaffNewsletterHubspotService);
+    doc = TestBed.get(DOCUMENT);
+    httpClient = TestBed.get(HttpClient);
+  });
+  afterEach(() => {
+    httpMock.verify();
+    document.cookie = 'hubspotutk=; expires=Thu, 01 Jan 1970 00:00:00 GMT'
+  });
+  it('should be created', () => {
+    expect(newsletterService).toBeTruthy();
+  });
+
+  describe('sending a newsletter submission to hubspot', () => {
+    it('should take Hubspot configuration from the module configuration', async(inject([HttpClient, DaffNewsletterConfig], (http, config) => {
+      spyOn(http, 'post').and.callThrough();
+      const submission = { email: 'email@email.com', firstName: 'John', lastName: 'Doe' };
+      const version = config.version === undefined ? 'v3' : config.version;
+      const submissionString = 'https://api.hsforms.com/submissions/' + version + '/integration/submit/'
+        + config.portalId + '/' + config.guid
+      newsletterService.send(submission).subscribe(() => {
+        expect(http.post).toHaveBeenCalledWith(submissionString, jasmine.any(Object));
+      });
+
+      const req = httpMock.expectOne('https://api.hsforms.com/submissions/v3/integration/submit/undefined/undefined');
+
+      req.flush('');
+    })));
+    it('should transform a request send to hubspot', async(inject([DaffNewsletterTransformer], (transformer) => {
+      spyOn(transformer, 'transformOut').and.callThrough();
+      const submission = { email: 'email@email.com', firstName: 'John', lastName: 'Doe' };
+      newsletterService.send(submission).subscribe(() => {
+        expect(transformer.transformOut).toHaveBeenCalled();
+      });
+      const req = httpMock.expectOne('https://api.hsforms.com/submissions/v3/integration/submit/undefined/undefined');
+
+      req.flush('');
+    })));
+    it('should transform the response recieved from Hubspot', async(inject([DaffNewsletterTransformer], (transformer) => {
```

Comment:
This shouldn't use async and inject. It should be injected like the other services via the TestBed.

Reply:
Addressed in 7aeb11b1ed22f3bb2f03cf9a1023156c2824ca5b